src/typings-new-format/spec-1.0/api.ts

```diff
@@ -118,31 +117,31 @@ export interface Overlay extends Api10 {
     usage?: string
 }
 
-export interface LibraryBase10 {
+export interface LibraryBase10 extends common.FragmentDeclaration {
 
     /**
      * Declarations of (data) types for use within this API
      */
-    types?: { [key: string]: datamodel.TypeDeclaration }
+    types?: datamodel.TypeDeclaration[]
 
     /**
      * Declarations of resource types for use within this API
      */
-    resourceTypes?: { [key: string]: resources.ResourceType10 }
+    resourceTypes?: resources.ResourceType10[]
```

Comment:
Infact, key equals name for components declared in the main file and 
```
key == namespace + '.' + name
```
for components defined in libraries. So, in the former case the result is different.

> Though, the resource type itself does not contain a name property in RAML spec.

In fact it does (quite implicitly):

> Resource types can be declared using the OPTIONAL resourceTypes node at the root of the API definition. The value of this node is a map where keys names become **names** of resource types that can be referenced throughout the API, and values are resource type declarations.